contracts/nexus-gateway/src/nexus.rs

```diff
@@ -22,6 +22,7 @@ pub struct Message {
     pub source_tx_id: nonempty::Vec<u8>,
     pub source_tx_index: u64,
     pub id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
So what does this mean for deserializing older versions of the struct, does it just deserialize to None? If so, can we test this?

Reply:
There's no "older" version cuz the field was just added in my previous PR. Yes it deserializes `null` or `undefined` or not exist to None. I am not sure if I should test it cuz I don't wanna test every single serde feature.

Reply:
why do we need this in the first place?

Reply:
It's not entirely needed. It's just silly to send extra data cross the VM and core.